### DISCUSSION TITLE
Theia Cloud base images

### DISCUSSION THREAD

**Discussion by Mtze** [eclipse-theia/theia-cloud#300] [Q&A]
Hi everyone, 

I'm currently tying to build customized Theia images for different use cases and ran into a couple of problems. 

Is there a minimal example for a theia image build (without java, lots of extension, etc) as a starting point? 

I'd love to start minimal and build up on that. The documentation [on your website](https://main--theia-cloud.netlify.app/documentation/addapplication/) is not working unfortunately (I already opened a bug in the website repo). Also the example image specs in the theia-blueprint repo are relatively bulky (1.2G - ~8 min build time) and even bundle the whole blueprint repository. There is also a missmatch in the node version between the example on the website and the Dockerfiles in the repo. 

Would you mind sharing some details on what a minimal theia image Dockerfile would look like? 
Thanks in advance! 
Matthias

**Comment by lucas-koehler** [comment]
Hi @Mtze , thanks for reporting the issue in the website repository. There, I linked a [PR](https://github.com/eclipsesource/theia-cloud-website/pull/30) to improve the documentation. The Dockerfile itself mentioned on the website itself is already quite minimal when not executed in the blueprint repository. You can use the [Theia yoeman template](https://github.com/eclipse-theia/generator-theia-extension) to scaffold a minimal Theia application. Note the image will still have a couple 100 MB even for a minimal application due to the size of the base image and Theia's dependencies.

Does this solve your question? If not, please feel free to pose follow up questions :)

**Reply by Mtze** [reply]
Where would be your prefered location (repository wise) for a minimal Dockerfile? blueprint? Maybe we can contribute here and also add some documentation from the outside perspective :)

**Reply by lucas-koehler** [reply]
Hi @Mtze , I think the best place to improve the minimal Dockerfile and add additional documentation is the documentation part of the [website](https://github.com/eclipsesource/theia-cloud-website). We want to avoid splitting adopter documentation across the website and the repositories. Furthermore, I think this is a better fit than the blueprint repository as it is not about a minimal application and not necessarily for hosting.
So you can gladly add to the [Add Application documentation](https://github.com/eclipsesource/theia-cloud-website/blob/main/content/documentation/addApplication/content.md) via a pull request and we will have a look. An alternative would be that you collect the minimal Dockerfile and documentation, send it to us and we integrate it into the documentation.

@jfaltermeier Would you agree with this assessment?

**Reply by jfaltermeier** [reply]
Yes, I agree. The Blueprint Repository will be renamed from theia-blueprint to theia-ide in the near future, so a minimal example won't really fit there.
